lib/spack/spack/directives.py

```diff
@@ -180,24 +180,43 @@ def _depends_on(pkg, spec, when=None):
         when = pkg.name
     when_spec = parse_anonymous_spec(when, pkg.name)
 
+    if deptypes is None:
+        # The default deptype is build and link because the common case is to
+        # build against a library which then turns into a runtime dependency
+        # due to the linker.
```

Comment:
`curl` and `git` are probably `build` dependencies?

I think for distant observer like me it is not clear what `run` dependencies are in the context of package manager like Spack? You don't normally use/run application using Spack? Maybe this could be clarified in the description of `run` at the top.